site/src/components/mdx/mdx-page-content.tsx

```diff
@@ -108,7 +108,7 @@ export const MdxPageContent: VFC<MdxPageContentProps> = ({
 
   useEffect(() => {
     const onScroll = () => {
-      if (!detectHeadingFromScroll) return;
+      if (!detectHeadingFromScroll || headings.length === 0) return;
 
       let headingAtScrollPosition: Heading = headings[0]!;
```

Comment:
Added this check since there were scenarios where `headings` was an empty array and that caused an error.